### Exercise 2

In [1]:
import requests
import json

# Setting  IBM Watson NLU API credentials and endpoint
apikey = 'qXBbo61tWIEYAPWk1fJByjbyCpfY0hZR3jNc6xvXEka9'
url = 'https://api.us-south.natural-language-understanding.watson.cloud.ibm.com/instances/f55ba5ba-4ca9-4ec2-a47b-d72fe26d7fc3/v1/analyze?version=2019-07-12'

# The provided text
text_to_analyze = """The United Kingdom suffered one of the worst recessions among major economies last year. Germany, for example, held up better in the pandemic than it did during the global financial crisis. Provisional estimates suggest Europe's biggest economy contracted by 5% last year. EU GDP, meanwhile, is expected to have shrunk 6.4%, according to Eurostat.
The United States fared even better by comparison, with GDP decreasing by 3.5% from the prior year.
"Today's figures show that the economy has experienced a serious shock as a result of the pandemic, which has been felt by countries around the world," UK finance minister Rishi Sunak said in a statement. "While there are some positive signs of the economy's resilience over the winter, we know that the current lockdown continues to have a significant impact on many people and businesses."
The new national lockdown in the United Kingdom, imposed on January 5, is expected to hit the economy hard in the first quarter of 2021, reversing the return to growth in the fourth quarter of 2020.
"It seems that a double dip [recession] was merely delayed rather than avoided outright," Sam Miley, an economist at the London-based Centre for Economics and Business Research said in a note on Friday.
Disruption to EU-UK trade following the end of the Brexit transition period on December 31 is also weighing on activity.”"""

# Setup the request payload
payload = {
    "text": text_to_analyze,
    "features": {
        "sentiment": {}
    }
}

# Making the POST request to the IBM Watson NLU API
response = requests.post(url, auth=('apikey', apikey), headers={'Content-Type': 'application/json'}, data=json.dumps(payload))

# Check if the request was successful and print the response
if response.status_code == 200:
    print(json.dumps(response.json(), indent=2))
else:
    print(f"Request failed with status code {response.status_code}: {response.text}")


{
  "usage": {
    "text_units": 1,
    "text_characters": 1363,
    "features": 1
  },
  "sentiment": {
    "document": {
      "score": -0.677186,
      "mixed": "1",
      "label": "negative"
    }
  },
  "language": "en"
}


The analyzed document exhibits a negative sentiment with a score of `-0.677186`, which indicates strong negative emotions. The "mixed" attribute, marked as `1`, suggests that the document's sentiment is complex and may contain mixed emotions or attitudes.

IBM's approach to sentiment analysis primarily aims to comprehend the underlying emotions, opinions, and sentiments expressed in text data. It achieves this through advanced natural language processing (NLP) techniques and machine learning models.
